src/main/java/net/rptools/maptool/client/ClientMessageHandler.java

```diff
@@ -341,8 +341,14 @@ private void handle(SetZoneVisibilityMsg msg) {
 
           var zone = MapTool.getCampaign().getZone(zoneGUID);
           zone.setVisible(visible);
-
           ZoneRenderer currentRenderer = MapTool.getFrame().getCurrentZoneRenderer();
+          List<MacroButtonProperties> AllMacroButtons = new ArrayList<>();
+          Collection<GUID> AllTokenIDs = new ArrayList<>();
+          for (Token token : renderer.getZone().getAllTokens()) {
+            AllTokenIDs.add(token.getId());
+          }
+          renderer.getSelectionModel().removeTokensFromSelection(AllTokenIDs);
+          MapTool.getFrame().setCurrentZoneRenderer(null);
```

Comment:
The above change should be in the below if statement (where it is setting the current zone renderer to null) because, at the moment, it will get executed for players and GMs when any map is made invisible or visible.

Reply:
Hi, i updated the commit to put the code below the if statement, now its working.